airthings_ble/parser.py

```diff
@@ -565,25 +597,49 @@ async def _get_service_characteristics(
                                 ),
                             )[0]
 
-                        device.sensors.update(
+                        sensors.update(
                             {
                                 "illuminance": command_sensor_data.get("illuminance"),
                                 "battery": bat_pct,
                             }
                         )
 
                     # Stop notification handler
-                    await client.stop_notify(characteristic.uuid)
+                    await client.stop_notify(characteristic)
 
-        return device
+    def _handle_disconnect(
+        self, disconnect_future: asyncio.Future[bool], client: BleakClient
+    ) -> None:
+        """Handle disconnect from device."""
+        self.logger.debug("Disconnected from %s", client.address)
+        disconnect_future.set_result(True)
```

Comment:
I sometimes get this error:
```
2024-01-19 00:42:19.338 ERROR (MainThread) [root] A message handler raised an exception: invalid state
Traceback (most recent call last):
  File "src/dbus_fast/message_bus.py", line 802, in dbus_fast.message_bus.BaseMessageBus._process_message
  File "/home/lastrada/airthings/core/venv/lib/python3.11/site-packages/bleak/backends/bluezdbus/manager.py", line 998, in _parse_msg
    watcher.on_connected_changed(new_connected)
  File "/home/lastrada/airthings/core/venv/lib/python3.11/site-packages/bleak/backends/bluezdbus/client.py", line 168, in on_connected_changed
    self._disconnected_callback()
  File "/home/lastrada/airthings/airthings-ble/airthings_ble/parser.py", line 615, in _handle_disconnect
    disconnect_future.set_result(True)
asyncio.exceptions.InvalidStateError: invalid state
```

Reply:
fixed in 400ef9b70ba58879c9f3d8586558dfc449020530